imagescan/delta_test.go

```diff
@@ -245,4 +245,136 @@ func TestDelta(t *testing.T) {
 		_, err = delta.findBestNode([]string{"node1"}, memQty.AsDec(), cpuQty.AsDec())
 		r.ErrorIs(err, errNoCandidates)
 	})
+
+	t.Run("frees up resources", func(t *testing.T) {
+		r := require.New(t)
+		delta := NewDeltaState(nil)
+
+		delta.upsert(&corev1.Node{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Node",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node1",
+			},
+			Status: corev1.NodeStatus{
+				Allocatable: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("2"),
+					corev1.ResourceMemory: resource.MustParse("4Gi"),
+				},
+			},
+		})
+
+		pod := &corev1.Pod{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				UID: "123",
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "test",
+						Image: "test",
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("1"),
+								corev1.ResourceMemory: resource.MustParse("2Gi"),
+							},
+						},
+					},
+				},
+				NodeName: "node1",
```

Comment:
`nitpick`: create node into variable so it could be reused here

Reply:
This could eventually get out of control. A little repetition didn't hurt anyone. Besides this could turn against us in the future when one of test cases will require some customizations.